src/modelmanager.cpp

```diff
@@ -543,6 +605,36 @@ Status ModelManager::reloadModelWithVersions(ModelConfig& config) {
     std::shared_ptr<model_versions_t> versionsToRetire;
 
     auto model = getModelIfExistCreateElse(config.getName());
+
+    // first set custom loader name to empty string so that any changes to name can be captured
+    std:: string emptyString = std::string();
+    model->setCustomLoaderName(emptyString);
```

Comment:
any non initialized string is "empty". Is this to clear previous custom loader name?

Reply:
Yes. This is to clear old name. Otherwise, I need to add another API to model class to clear the name.

Reply:
It would be clearer to have a method in model (fetchCustomLoaderName) that would handle all the logic required for this - in case we have custom loader set the name/ if not set this as empty.